spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/internal/spring/ArmeriaConfigurationUtil.java

```diff
@@ -124,7 +124,8 @@ public static void configureServerWithArmeriaSettings(
         final boolean needsManagementPort =
                 findBean(beanFactory, "actuatorServerConfigurator", ArmeriaServerConfigurator.class) != null;
 
-        if (needsManagementPort && managementServerPort != null) {
+        if (needsManagementPort && managementServerPort != null &&
+            (internalServicePort == null || internalServicePort.getPort() != managementServerPort.getPort())) {
             internalPortsBuilder.add(managementServerPort);
         }
```

Comment:
I am considering how to test the modified code. After creating the `Server` through `Server#build()`, duplicate ports are eliminated by `ServerBuilder#resolveDistinctPorts`, and duplicate routes are also removed by `Routers.resolveAmbiguousRoutes()`, making it difficult to test effectively.

As a solution, I tried to test the state of the `ServerBuilder` before building it. However, I encountered an issue since the `ports` and `virtualHostBuilders` fields inside the `ServerBuilder` instance are private, which prevents me from testing them directly. I am thinking of making these two fields public and adding `@VisibleForTesting` annotation. Would it be acceptable to do so?

The expected test code is as follows:

```java
ServerBuilder serverBuilder = Server.builder();
ArmeriaConfigurationUtil.configureServerWithArmeriaSettings(serverBuilder, ... )
// Testing whether there are any duplicate ports in `serverBuilder.ports`.
// Testing whether there are any duplicates in serviceConfigSetters inside the VirtualHostBuilder.
```

Reply:
I think it's okay not to have a test for this case because adding the same port to the list was a mistake obviously.
Thanks for considering such a solution by the way. 👍